js/apps/admin-ui/src/realm-settings/RealmSettingsTabs.tsx

```diff
@@ -235,6 +235,34 @@ export const RealmSettingsTabs = () => {
         ),
       );
     }
+    if (r.attributes) {
+      // Handle nested vc object created by React Hook Form
+      if (r.attributes["vc"] && typeof r.attributes["vc"] === "object") {
+        const vcObj = r.attributes["vc"] as Record<string, any>;
+        Object.keys(vcObj).forEach((key) => {
+          const flatKey = `vc.${key}`;
+          r.attributes![flatKey] = vcObj[key]?.toString() || vcObj[key];
+        });
+        // Remove the nested object
+        delete r.attributes["vc"];
+      }
+
+      // Convert OID4VCI attributes to strings if they're not already
+      if (
+        r.attributes["vc.c-nonce-lifetime-seconds"] !== undefined &&
+        typeof r.attributes["vc.c-nonce-lifetime-seconds"] !== "string"
+      ) {
+        r.attributes["vc.c-nonce-lifetime-seconds"] =
+          r.attributes["vc.c-nonce-lifetime-seconds"].toString();
+      }
+      if (
+        r.attributes["preAuthorizedCodeLifespanS"] !== undefined &&
+        typeof r.attributes["preAuthorizedCodeLifespanS"] !== "string"
+      ) {
+        r.attributes["preAuthorizedCodeLifespanS"] =
+          r.attributes["preAuthorizedCodeLifespanS"].toString();
+      }
```

Comment:
Rather than checking all the things a value might 'not' be, perhaps it would be easier to check for the type you are actually expecting? Also, it might be cleaner to handle this is a small loop with all the attributes pre-listed, to prevent repetition:

```suggestion
      const stringifiedAttributes = [
        "vc.c-nonce-lifetime-seconds",
        "preAuthorizedCodeLifespanS"
      ];

      for (const attr of stringifiedAttributes) {
        const value = r.attributes[attr];

        if (typeof value === "number") {
          r.attributes[attr] = value.toString();
        }
      }
```